src/unix/process.c

```diff
@@ -963,6 +983,74 @@ static int uv__spawn_and_init_child(
 }
 #endif /* ISN'T TARGET_OS_TV || TARGET_OS_WATCH */
 
+int uv__pty_resize_fd(int pty_fd,
+                  unsigned short cols,
+                  unsigned short rows) {
+  struct winsize winp;
+  winp.ws_col = cols;
+  winp.ws_row = rows;
+  winp.ws_xpixel = 0;
+  winp.ws_ypixel = 0;
+  if (ioctl(pty_fd, TIOCSWINSZ, &winp) < 0)
+    return UV__ERR(errno);
+  return 0;
+}
+
+int uv__spawn_make_pty(int *fd_pty, int *fd_tty, int cols, int rows) {
+    int ret;
+    int my_errno;
+
+    *fd_pty = posix_openpt(O_RDWR);
+    if (*fd_pty < 0)
+        return UV__ERR(errno);
+
+    if (grantpt(*fd_pty) < 0) {
+        my_errno = UV__ERR(errno);
+        close(*fd_pty);
+        return my_errno;
+    }
+
+    if (unlockpt(*fd_pty) < 0) {
+        my_errno = UV__ERR(errno);
+        close(*fd_pty);
+        return my_errno;
+    }
+
+    int path_tty_size = 40;
+    char *path_tty = uv__malloc(path_tty_size * sizeof(char));
```

Comment:
The realloc loop isn't really necessary, you can declare a `char path_tty[TTY_NAME_MAX + 1] = {0};` buffer on the stack.